src/tests/mempool-tests.ts

```diff
@@ -1405,4 +1405,125 @@ describe('mempool tests', () => {
     expect(mempoolTxIdsAfterReOrg).toEqual(['0x04bb', '0x05bb']);
   });
+
+  test('Reconcile mempool pruned status', async () => {
+    const senderAddress = 'SP25YGP221F01S9SSCGN114MKDAK9VRK8P3KXGEMB';
+    const txId = '0x521234';
+    const dbBlock1: DbBlock = {
+      block_hash: '0x0123',
+      index_block_hash: '0x1234',
+      parent_index_block_hash: '0x5678',
+      parent_block_hash: '0x5678',
+      parent_microblock_hash: '',
+      parent_microblock_sequence: 0,
+      block_height: 1,
+      burn_block_time: 39486,
+      burn_block_hash: '0x1234',
+      burn_block_height: 123,
+      miner_txid: '0x4321',
+      canonical: true,
+      execution_cost_read_count: 0,
+      execution_cost_read_length: 0,
+      execution_cost_runtime: 0,
+      execution_cost_write_count: 0,
+      execution_cost_write_length: 0,
+    };
+    const dbBlock2: DbBlock = {
+      block_hash: '0x2123',
+      index_block_hash: '0x2234',
+      parent_index_block_hash: dbBlock1.index_block_hash,
+      parent_block_hash: dbBlock1.block_hash,
+      parent_microblock_hash: '',
+      parent_microblock_sequence: 0,
+      block_height: 2,
+      burn_block_time: 39486,
+      burn_block_hash: '0x1234',
+      burn_block_height: 123,
+      miner_txid: '0x4321',
+      canonical: true,
+      execution_cost_read_count: 0,
+      execution_cost_read_length: 0,
+      execution_cost_runtime: 0,
+      execution_cost_write_count: 0,
+      execution_cost_write_length: 0,
+    };
+    const mempoolTx: DbMempoolTxRaw = {
+      tx_id: txId,
+      anchor_mode: 3,
+      nonce: 0,
+      raw_tx: bufferToHexPrefixString(Buffer.from('test-raw-mempool-tx')),
+      type_id: DbTxTypeId.Coinbase,
+      status: 1,
+      post_conditions: '0x01f5',
+      fee_rate: 1234n,
+      sponsored: false,
+      sponsor_address: undefined,
+      sender_address: senderAddress,
+      origin_hash_mode: 1,
+      coinbase_payload: bufferToHexPrefixString(Buffer.from('hi')),
+      pruned: false,
+      receipt_time: 1616063078,
+    };
+    const dbTx1: DbTxRaw = {
+      ...mempoolTx,
+      ...dbBlock1,
+      parent_burn_block_time: 1626122935,
+      tx_index: 4,
+      status: DbTxStatus.Success,
+      raw_result: '0x0100000000000000000000000000000001', // u1
+      canonical: true,
+      microblock_canonical: true,
+      microblock_sequence: I32_MAX,
+      microblock_hash: '',
+      parent_index_block_hash: '',
+      event_count: 0,
+      execution_cost_read_count: 0,
+      execution_cost_read_length: 0,
+      execution_cost_runtime: 0,
+      execution_cost_write_count: 0,
+      execution_cost_write_length: 0,
+    };
+
+    // Simulate the bug with a txs being in the mempool at confirmed at the same time by
+    // directly inserting the mempool-tx and mined-tx, bypassing the normal update functions.
+    await db.updateBlock(db.sql, dbBlock1);
+    await db.updateMempoolTxs({ mempoolTxs: [mempoolTx] });
```

Comment:
If we apply the fix to `updateMempoolTxs` too, you could still run this test just by inserting the tx directly into the DB here. Or maybe reversing the order of these two lines.

Reply:
Good idea, done!